### DISCUSSION TITLE
Charge state corresponding to ions

### DISCUSSION THREAD

**Discussion by phy-ak** [epochpic/epoch#558] [General]
As mentioned in the newly upgraded EPOCH 4.19 ionisation pdf,

The first number on each line describes the ionisation energy from neutral to charge state 1+, the second for 1+ to 2+, etc.

 Consider when we mention charge as 2 corresponding to carbon (C2+), when we visualise in VisIt, carbon ionises to Carbon1, Carbon2, Carbon3, Carbon4 whereas C5 and C6 is removed. 

But when we compare it with the ionisation energy , Carbon1 and Carbon2 should be removed accordingly instead of Carbon5 and Carbon6.

 How to understand this? Also how the number density and particle energy should vary based on different charge states as  I don't find any trend between different charge states? 

Attached here is the example deck

begin:species
name = Carbon
mass = 12.0 1836.2
charge = 2.0
atomic_no = 6
ionise = T
density = if((x gt 2.0micron) and (x lt 3.0*micron), den_c, 0.0)
npart_per_cell = 32
ionisation_electron_species = Electron
end:species

begin:species
name = Electron
charge = -1.0
mass = 1.0
density = if((x gt 2.0micron) and (x lt 3.0micron), 2*den_c, 0.0)
npart_per_cell = 32
end:species

**Comment by Status-Mirror** [comment]
Hi @Arun-lime 

Consider your block:

```
begin:species
  name = Carbon
  mass = 12.0 * 1836.2
  charge = 2.0
  atomic_no = 6
  ionise = T
  density = if((x gt 2.0micron) and (x lt 3.0*micron), den_c, 0.0)
  npart_per_cell = 32
  ionisation_electron_species = Electron
end:species
```

EPOCH will read `charge = 2.0`, and `atomic_no = 6`, and then it knows you have Carbon-2. EPOCH knows your species has a charge-state of 2, so the first thing it does is check to see if your species name ends in a 2. In your case, you use the species name `Carbon`, which does not end in a 2. Hence, the `Carbon` species has a charge of 2, the `Carbon` species ionises to the `Carbon1` species which has a charge of 3, then up to `Carbon4` which has a charge of 6. EPOCH always names an ionised state with a number which is one higher than the previous.

If you had instead used the species name `Carbon2`, then EPOCH would see that you have a species name which ends in a 2, describing an ionising species with a charge of 2. Only then will EPOCH take `Carbon` as the base-state, and create a species called `Carbon3` for charge-state 3.

Does this make sense?
Stuart
